### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by CDobbins1**

### CODE REVIEW

**Comment by twofingerrightclick** [comment]
Will review.

File: SecretSanta/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// This file is used by Code Analysis to maintain SuppressMessage
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given
+// a specific target and scoped to a namespace, type, member, etc.
+
```
Comment (twofingerrightclick) [inline SecretSanta/GlobalSuppressions.cs]: reasonable justifications

File: SecretSanta/TestSuppressions.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿// This file is used by Code Analysis to maintain SuppressMessage
```
Comment (twofingerrightclick) [inline SecretSanta/TestSuppressions.cs]: I like the TestSuppressions naming

File: SecretSanta/src/SecretSanta.Business/Gift.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        private string _Title = "<Invalid>";
+        public string Title
+        {
+            get => _Title;
+            set => _Title = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private string _Description = "<Invalid>";
+        public string Description
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Business/Gift.cs]: Good implementation of data validation in the setter versus constructor

File: SecretSanta/test/SecretSanta.Business.Tests/GiftTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        public void CreateGift_UsingNullTitle_ThrowsArgumentNullException()
+        {
+            Gift gift = new Gift(1, null!, "TestDescription", "TestUrl", new User(1, "TestFirst", "TestLast", new List<Gift>()));
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ChangeGiftTitle_UsingNullTitle_ThrowsArgumentNullException()
+        {
+            Gift gift = new Gift(1, "TestTitle", "TestDescription", "TestUrl", new User(1, "TestFirst", "TestLast", new List<Gift>()));
```
Comment (twofingerrightclick) [inline SecretSanta/test/SecretSanta.Business.Tests/GiftTests.cs]: Just an api implementation comment. Do we want to always pass a list of gifts to the User Constructor, or have an option where its constructor creates the gift list, and a setter if we want to pass an already populated gift list.

File: SecretSanta/src/SecretSanta.Business/Gift.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        private string _Url = "<Invalid>";
+        public string Url
+        {
+            get => _Url;
+            set => _Url = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        public User? User { get; set; }
```
Comment (twofingerrightclick) [inline SecretSanta/src/SecretSanta.Business/Gift.cs]: can be null. I think that makes sense

File: SecretSanta/test/SecretSanta.Business.Tests/GiftTests.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            Assert.AreEqual<string>("TestDescription", gift.Description);
+            Assert.AreEqual<string>("TestUrl", gift.Url);
+            Assert.IsNotNull(gift.User);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void CreateGift_UsingNullTitle_ThrowsArgumentNullException()
+        {
+            Gift gift = new Gift(1, null!, "TestDescription", "TestUrl", new User(1, "TestFirst", "TestLast", new List<Gift>()));
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Business.Tests/GiftTests.cs]: In these cases you can use throw away variables so the compilers will not allocate space for them.
```suggestion
            var _ = new Gift(1, null!, "TestDescription", "TestUrl", new User(1, "TestFirst", "TestLast", new List<Gift>()));
```

File: Assignment1.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Assignment
```
Comment (Keboo) [inline Assignment1.md]: Seeing files like this in the PR indicates that your `Assignment1` branch was not based on the IntelliTect Assignment1 branch. Please review the directions [here](https://github.com/IntelliTect-Samples/EWU-CSCD379-2020-Winter/wiki/Homework-Assignments#creating-an-assignment-branch) and let me know if you still have questions on creating assignment branches.

File: SecretSanta/src/SecretSanta.Business/User.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        private string _LastName = "<Invalid>";
+        public string LastName
+        {
+            get => _LastName;
+            set => _LastName = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        public List<Gift> Gifts { get; }
+
+        public User(int id, string firstName, string lastName, List<Gift> gifts = null!)
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Business/User.cs]: You should not use `null!` here as the default value. If you want to allow it the parameter should have been declared as nullable.
```suggestion
        public User(int id, string firstName, string lastName, List<Gift>? gifts = null)
```

**Comment by CDobbins1** [comment]
@Keboo Here is the link to the pipeline
https://dev.azure.com/cdobsc01/CSCD379-2020-Winter

**Comment by Keboo** [comment]
@CDobbins1 thanks, I have updated it.

**Pull request closed without merging**